src/AppInstallerCLICore/Workflows/UninstallFlow.cpp

```diff
@@ -0,0 +1,201 @@
+
+            // Wait for uninstallation to finish
+            while (!progress.IsCancelled())
+            {
+                DWORD waitResult = WaitForSingleObject(process.get(), 250);
+                if (waitResult == WAIT_OBJECT_0)
+                {
+                    break;
+                }
+
+                if (waitResult != WAIT_TIMEOUT)
+                {
+                    THROW_LAST_ERROR_MSG("Unexpected WaitForSingleObjectResult: %d", waitResult);
+                }
+            }
+
+            if (progress.IsCancelled())
+            {
+                return {};
+            }
+            else
+            {
+                DWORD exitCode = 0;
+                GetExitCodeProcess(process.get(), &exitCode);
+                return exitCode;
+            }
+        }
+    }
+
+    void GetUninstallInfo(Execution::Context& context)
+    {
+        auto installedPackageVersion = context.Get<Execution::Data::InstalledPackageVersion>();
+        const std::string installedTypeString = installedPackageVersion->GetMetadata()[PackageVersionMetadata::InstalledType];
+        switch (ManifestInstaller::ConvertToInstallerTypeEnum(installedTypeString))
+        {
+        case ManifestInstaller::InstallerTypeEnum::Exe:
+        case ManifestInstaller::InstallerTypeEnum::Burn:
+        case ManifestInstaller::InstallerTypeEnum::Inno:
+        case ManifestInstaller::InstallerTypeEnum::Msi:
+        case ManifestInstaller::InstallerTypeEnum::Nullsoft:
+        case ManifestInstaller::InstallerTypeEnum::Wix:
+        {
+            IPackageVersion::Metadata packageMetadata = installedPackageVersion->GetMetadata();
+            IPackageVersion::Metadata::iterator itr;
+
+            if (context.Args.Contains(Execution::Args::Type::Interactive))
+            {
+                itr = packageMetadata.find(PackageVersionMetadata::StandardUninstallCommand);
+            }
+            else if (context.Args.Contains(Execution::Args::Type::Silent))
+            {
+                itr = packageMetadata.find(PackageVersionMetadata::SilentUninstallCommand);
+            }
+            else
+            {
+                itr = packageMetadata.find(PackageVersionMetadata::StandardUninstallCommand);
+                if (itr == packageMetadata.end())
+                {
+                    itr = packageMetadata.find(PackageVersionMetadata::SilentUninstallCommand);
+                }
+            }
+
+            if (itr == packageMetadata.end())
+            {
+                AICLI_TERMINATE_CONTEXT(E_ABORT);
+            }
+            context.Add<Execution::Data::UninstallString>(itr->second);
+
+            break;
+        }
+        case ManifestInstaller::InstallerTypeEnum::Msix:
+        case ManifestInstaller::InstallerTypeEnum::MSStore:
+        {
+            context.Add<Execution::Data::PackageFamilyNames>(installedPackageVersion->GetMultiProperty(PackageVersionMultiProperty::PackageFamilyName));
+            break;
```

Comment:
This should check for the case where no values are present and fail similarly to no uninstall string.